#75: Straight-forward implementation of GCP Secret Manager support

madsop opened the pull request:
For my project, I needed Secret Manager support for Quarkus native, so I decided to try to figure it out myself.

This is an implementation heavily inspired by the pubsub- and firestore-implementation. It's also an implementation that's very close to the original Google client library in user experience.

So I'm drafting this to hear your thoughts on it. Totally understand if it's too early.

loicmathieu commented:
I added a comment on the other PR to ping the contributor, if he didn't respond we should proceed with this one

loicmathieu commented:
So, #3 is abandonned and can be used as an inspiration.
I'm OK to proceed with this one, I'll review it as soon as it's ready.

madsop commented:
@loicmathieu , @dzou : I think this is ready for review now. I've added a readme file. I've also looked through the #3 PR and think that I've addressed the relevant comments from that one as well.

loicmathieu commented on secret-manager/README.md:
````diff
@@ -0,0 +1,55 @@
+</dependency>
+```
+
+## Some example
````
Please provides a Java example as all other example are in Java.

madsop replied:
Indeed, it's nice to be consistent. However, I think it's nice that the project has an example or two in Kotlin as well, for those (me included) who are using Kotlin with Quarkus native, just to prove that this project also can be helpful in that context. I added both a Java and a Kotlin version in the newly pushed commit, perhaps that's a suitable solution?

loicmathieu replied:
The issue here is consistency as this will be the only guide that incldues Kotlin example.
So, if you plan to contribute Kotlin example for the other guide, I'm OK with it (I'll love to see more Kotlin with Quarkus), otherwise it is better to avoid it.

The readability also is a concern, when using mutiple example on the same thing on different languages I expect the design to only show one language and having way to toggle from one to the other, we didn't have this currently ...

madsop replied:
Yeah, I see the concerns you're pointing out. I guess most people using Kotlin these days also know Java pretty well, and are able to convert a Java example into Kotlin - and if not, IntelliJ does that pretty good as well.

loicmathieu commented on secret-manager/README.md:
```diff
@@ -0,0 +1,55 @@
+# Quarkiverse - Google Cloud Services - Secret Manager
+
+This extension allows to inject the Google Cloud Platform Secret Manager client library inside your Quarkus application.
```
This is not "inject" but "use" as inject is connoted for CDI

loicmathieu reviewed (commented):
Thanks

I add some documentation related comments. 

Can you also add in the documentation the step to create the secret used in your example? 

I wonder how it works regarding authentication? It didn't use the gcp credentials provided  by the common lib.

loicmathieu commented:
Just saw that auth is enhanced on the PR from @dzou so this part is OK for this PR 👌

madsop commented:
> Thanks
> 
> I add some documentation related comments.
> 
> Can you also add in the documentation the step to create the secret used in your example?
> 
> I wonder how it works regarding authentication? It didn't use the gcp credentials provided by the common lib.

Thanks for providing feedback! Just pushed an updated version of the readme file.

Also thanks to @dzou for improving the auth part, which I didn't put that much focus on. I've been using the GOOGLE_APPLICATION_CREDENTIALS environment variable locally, so that way it has been working on my machine.

loicmathieu reviewed (approved):
LGTM,
Thanks a lot!
I'll merge soon, I want to test it locally before merging it.

loicmathieu commented:
@all-contributors please add @madsop for code and doc
